### DISCUSSION TITLE
Does skywalking support IPV6

### DISCUSSION THREAD

**Discussion by dustyewer** [apache/skywalking#10826] [Ideas]
Does skywalking support IPV6？
Can java programs running in ipv6-only environments use skywalking-java to connect to skywalking in ipv4 environments?

**Comment by wu-sheng** [comment]
SkyWalking don't have self implementation about networkpart.
This relies on gRPC, Armeria and Kafka implementations.

**Comment by dustyewer** [comment]
Does the following code for the read configuration of skywalking agent not consider an IPV6 scenario?
GRPCChannelManager.java

                        String[] ipAndPort = server.split(":");

                        if (managedChannel != null) {
                            managedChannel.shutdownNow();
                        }

                        managedChannel = GRPCChannel.newBuilder(ipAndPort[0], Integer.parseInt(ipAndPort[1]))

**Reply by wu-sheng** [reply]
This is for separating ip and port only. What is the issue?

**Reply by dustyewer** [reply]
Ipv6 also uses : as a delimiter

**Reply by Superskyyy** [reply]
@wu-sheng @dustyewer I think this is relevant, the current code didn't take ipv6 addresses into consideration and will fail on it. Ipv6 addresses will look like `[2001:0db8:85a3:0000:0000:8a2e:0370:7334]:443`

**Reply by wu-sheng** [reply]
If format is different, PR to patch is welcome.
But honestly, I don't have any experience on IPv6 and neither have test env.

**Reply by Superskyyy** [reply]
> If format is different, PR to patch is welcome. But honestly, I don't have any experience on IPv6 and neither have test env.

I can implement this. As you said though, I got no way to properly test it, I can only write some unit tests for the parser.

**Reply by wu-sheng** [reply]
I hope we have env to verify. The codes are easy, I don't want to create confusion for this.

**Reply by Superskyyy** [reply]
@dustyewer to understand this better, you were saying skywalking in ipv4 env and agent in ipv6, it wouldn't work right? they both have to be in ipv6 addresses to establish connections. (This doesn't relate to code, the only code that needs to be changed is in the agents to allow ipv6 addresse configuration.

**Reply by dustyewer** [reply]
I have verified that after a simple change to the agent code (simply using "/" instead of ":" to split the ip and port), the agent can use the ipv6 address to connect to the server.

**Reply by dustyewer** [reply]
I encountered more problems when trying to deploy the server side in an ipv6 environment. I temporarily deployed the server and es in the docker container environment of ipv4 to bypass these problems.
